### PR TITLE
iOS-2591 DI | Migrate link to object 🤪 (migrate search P1)

### PR DESCRIPTION
**Pull request by mgolovko**

### CODE REVIEW

File: Anytype/Sources/PresentationLayer/Common/SwiftUI/Search/LinkToObject/LinToObjectSearchView.swift [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import Foundation
+import SwiftUI
+
+struct LinToObjectSearchView: View {
```
Comment (joe-pusya) [inline Anytype/Sources/PresentationLayer/Common/SwiftUI/Search/LinkToObject/LinToObjectSearchView.swift:4]: LinToObjectSearchView -> LinkToObjectSearchView

File: Anytype/Sources/PresentationLayer/Common/SwiftUI/Search/LinkToObject/LinkToObjectSearchViewModel.swift [hunk 2]
```diff
@@ -15,46 +15,71 @@ final class LinkToObjectSearchViewModel: SearchViewModelProtocol {
...
+                if let linkBlockDetails = try? await self?.defaultObjectService.createDefaultObject(name: name, shouldDeleteEmptyObject: false, spaceId: data.spaceId) {
+                    AnytypeAnalytics.instance().logCreateObject(objectType: linkBlockDetails.analyticsType, route: .mention)
+                    data.setLinkToObject(linkBlockDetails.id)
+                }
+            }
+        case let .web(url):
+            data.setLinkToUrl(url)
+        case let .openURL(url):
+            data.willShowNextScreen?()
+//            urlOpener.openUrl(url)
```
Comment (joe-pusya) [inline Anytype/Sources/PresentationLayer/Common/SwiftUI/Search/LinkToObject/LinkToObjectSearchViewModel.swift:69]: do we need this comments?

**Review by joe-pusya — APPROVED** [review]

File: Anytype/Sources/PresentationLayer/Common/SwiftUI/Search/LinkToObject/LinkToObjectSearchViewModel.swift (hunk 2 above)
Comment (mgolovko) [reply Anytype/Sources/PresentationLayer/Common/SwiftUI/Search/LinkToObject/LinkToObjectSearchViewModel.swift:69]: Fixed in next pr

File: Anytype/Sources/PresentationLayer/Common/SwiftUI/Search/LinkToObject/LinToObjectSearchView.swift (hunk 1 above)
Comment (mgolovko) [reply Anytype/Sources/PresentationLayer/Common/SwiftUI/Search/LinkToObject/LinToObjectSearchView.swift:4]: Renamed in next pr

**Pull request merged**